sdk/core/core-rest-pipeline/src/fetchHttpClient.ts

```diff
@@ -103,18 +101,23 @@ async function makeRequest(request: PipelineRequest): Promise<PipelineResponse>
     if (isBlob(request.body) && request.onUploadProgress) {
       request.onUploadProgress({ loadedBytes: request.body.size });
     }
-    return buildPipelineResponse(response, request);
-  } finally {
+    return buildPipelineResponse(response, request, abortControllerCleanup);
+  } catch (e) {
     if (abortControllerCleanup) {
       abortControllerCleanup();
     }
```

Comment:
nitpick: could be written as one liner:
```suggestion
    abortControllerCleanup?.()
```